src/data/climate.ts

```diff
@@ -40,23 +40,25 @@ export type ClimateEntity = HassEntityBase & {
     target_humidity_high?: number;
     min_humidity?: number;
     max_humidity?: number;
-    fan_mode?: FanMode;
-    fan_modes?: FanMode[];
-    preset_mode?: PresetMode;
-    preset_modes?: PresetMode[];
+    fan_mode?: string;
+    fan_modes?: string[];
+    preset_mode?: string;
+    preset_modes?: string[];
```

Comment:
why was this typing made less precise?

Reply:
It's not less precise. Preset and fan mode can be custom things so the enum doesn't make sense.